pathio_test.go

```diff
@@ -190,3 +196,78 @@ func TestS3FileWriterSuccessNoEncryption(t *testing.T) {
 	assert.Equal(t, foundErr, nil)
 	svc.AssertExpectations(t)
 }
+
+func TestS3ListFiles(t *testing.T) {
+	svc := mockedS3Client{}
+	bucket, key := "bucket", "key"
+	output := s3.ListObjectsOutput{
+		Contents: []*s3.Object{
+			&s3.Object{Key: aws.String("file1")},
+		},
+		CommonPrefixes: []*s3.CommonPrefix{
+			&s3.CommonPrefix{Prefix: aws.String("prefix/")},
+		},
+		IsTruncated: aws.Bool(false),
+	}
+
+	params := s3.ListObjectsInput{
+		Bucket:    aws.String(bucket),
+		Prefix:    aws.String(key),
+		Delimiter: aws.String("/"),
+	}
+
+	svc.On("ListObjects", &params).Return(&output, nil)
+	files, err := lsS3(s3Connection{&svc, bucket, key})
+	assert.NoError(t, err)
+	assert.True(t, reflect.DeepEqual([]string{"prefix/", "file1"}, files))
```

Comment:
`assert` isn't standard library, it uses `reflect` under the hood. Since we're already using the `assert` abstraction it's better to just stick with the higher level abstraction and not use both IMO